cpp/src/common/logger.cpp

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
I have separated declarations and definitions so as to limit spdlog dependency to only logger.cpp. This also helps our prims to remain header-only. But now, the prims unit-tests and benchmarks will have to add logger.cpp into their compilation sources in order to find spdlog during linker phase.